gtsfm/loader/argoverse_dataset_loader.py

```diff
@@ -0,0 +1,141 @@
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+RING_CAMERA_FRAME_RATE = 30  # fps or Hz
+
+
+class ArgoverseDatasetLoader(LoaderBase):
+    """ """
+
+    def __init__(
+        self,
+        dataset_dir: str,
+        log_id: str,
+        stride: int = 10,
+        max_num_imgs: int = 20,
+        max_lookahead_sec: float = 2,
+        camera_name: str = "ring_front_center",
+    ) -> None:
+        """
+        Args:
+            dataset_dir: directory where raw Argoverse logs are stored on disk
+            log_id: unique ID of vehicle log
+            stride: sampling rate, e.g. every 2 images, every 4 images, etc.
+            max_num_imgs: number of images to load (starting from beginning of log sequence)
+        """
+        self.log_id_ = log_id
+        self.dl_ = SimpleArgoverseTrackingDataLoader(data_dir=dataset_dir, labels_dir=dataset_dir)
+
+        # computed as #frames = #sec * (frame/sec)
+        max_lookahead_num_fr = max_lookahead_sec * RING_CAMERA_FRAME_RATE
+
+        # in subsampled list, account for subsampling rate
+        self.max_lookahead_for_img_ = max_lookahead_num_fr / stride
+
+        self.calib_data_ = self.dl_.get_log_calibration_data(log_id)
+
+        image_paths = self.dl_.get_ordered_log_cam_fpaths(log_id, camera_name)
+        image_paths = image_paths[::stride]
+        self.image_paths_ = image_paths[:max_num_imgs]
+
+        # for each image, cache its associated timestamp
+        self.image_timestamps_ = [int(Path(path).stem.replace(f"{camera_name}_", "")) for path in self.image_paths_]
+
+        cam_config = get_calibration_config(self.calib_data_, camera_name)
+        self.K_ = cam_config.intrinsic[:3, :3]
+
+        self.camera_SE3_egovehicle_ = SE3(
+            rotation=cam_config.extrinsic[:3, :3], translation=cam_config.extrinsic[:3, 3]
+        )
+        self.egovehicle_SE3_camera_ = self.camera_SE3_egovehicle_.inverse()
+
+    def __len__(self) -> int:
+        """The number of images in the dataset.
+
+        Returns:
+            The number of images.
+        """
+        return len(self.image_paths_)
+
+    def get_image(self, index: int) -> Image:
+        """Get the image at the given index.
+
+        Args:
+            index: the index to fetch.
+
+        Raises:
+            IndexError: if an out-of-bounds image index is requested.
+
+        Returns:
+            The image at the query index.
+        """
+
+        if index < 0 or index >= self.__len__():
+            raise IndexError("Image index is invalid")
+
+        return io_utils.load_image(self.image_paths_[index])
+
+    def get_camera_intrinsics(self, index: int) -> Optional[Cal3Bundler]:
+        """Get the camera intrinsics at the given index.
+
+        Args:
+            the index to fetch.
+
+        Returns:
+            Intrinsics for the given camera.
+        """
+        return Cal3Bundler(
+            fx=min(self.K_[0, 0], self.K_[1, 1]),
```

Comment:
can we add a comment saying they are the same thing. The `min` logic will confuse us.

Reply:
sure, good idea